crawl-ref/source/mon-act.cc

```diff
@@ -2222,6 +2222,63 @@ static void _torpor_snail_slow(monster* mons)
     }
 }
 
+/**
+ *  Make this monster use the magic of a holy lantern. (Vaud).
+ *
+ *  @param mons       The monster holding the lantern.
+ */
+static void _vaud_lantern_burn(monster *mons)
+{
+    //hurt player
+    //check in range and in LOS
+    if (cell_see_cell(you.pos(), mons->pos(), LOS_SOLID_SEE)
+        && grid_distance(mons->pos(), you.pos()) < 4)
+    {
+        //check divine protection
+        if (!is_good_god(you.religion)
+            && !is_sanctuary(you.pos()))
```

Comment:
For those occasions on which you've invoked Zin's sanctuary and then been excommunicated immediately...?